src/main/java/uk/gov/hmcts/cmc/claimstore/services/pilotcourt/PilotCourtService.java

```diff
@@ -144,15 +152,38 @@ private boolean checkPilotCourt(PilotCourt pilotCourt, String courtName, Pilot p
         return pilotCourts;
     }
 
-    private Set<Pilot> getPilots(CSVRecord next) {
-        Set<Pilot> pilots = new HashSet<>();
-        if (Boolean.parseBoolean(next.get(PilotCourtCSVHeader.LA_PILOT.ordinal()))) {
-            pilots.add(Pilot.LA);
-        }
-        if (Boolean.parseBoolean(next.get(PilotCourtCSVHeader.JDDO_PILOT.ordinal()))) {
-            pilots.add(Pilot.JDDO);
+    private Map<Pilot, LocalDateTime> getPilots(CSVRecord csvRecord) {
+        return Arrays.stream(PilotCourtCSVHeader.values())
+        .filter(pilotCourtCSVHeader -> pilotCourtCSVHeader.getPilot().isPresent())
+        .map(pilotCourtCSVHeader -> {
+            String pilotConfig = csvRecord.get(pilotCourtCSVHeader.ordinal());
+
+            if (!isValidPilotDate(pilotConfig)) {
+                return null;
+            }
+
+            return new Object[]{
+                pilotCourtCSVHeader.getPilot().get(),
+                LocalDate.parse(pilotConfig, ISO_LOCAL_DATE).atTime(PILOT_COURT_GO_LIVE_TIME, 0)
```

Comment:
Eek. I think if we have to resort to arbitrary structures like this it might be clearer using pre-8 language
```java
        final Map<Pilot, LocalDateTime> results = new HashMap<>();
        for (PilotCourtCSVHeader header : PilotCourtCSVHeader.values()) {
            Optional<Pilot> pilot = header.getPilot();
            if (!pilot.isPresent()) {
                continue;
            }
            String pilotConfig = csvRecord.get(header.ordinal());
            if (!isValidPilotDate(pilotConfig)) {
                continue;
            }
            results.put(
                pilot.get(),
                LocalDate.parse(pilotConfig, ISO_LOCAL_DATE).atTime(PILOT_COURT_GO_LIVE_TIME, 0)
            );
        }
        return results;
```